### DISCUSSION TITLE
How to hide marker of another side

### DISCUSSION THREAD

**Discussion by ekaterina4952** [pmndrs/drei#1197] [Q&A] [answered]
We are trying add a marker as dynamically in model. The marker is a html.
But this marker shows in all side even though model is rotated. We hope to hide if model shows other side that has nothing marker.

![image](https://user-images.githubusercontent.com/87656984/209636203-4fdf0b48-7472-4a38-9324-110d7d7e513d.png) ![image](https://user-images.githubusercontent.com/87656984/209636301-bfe21705-017f-4324-bafd-9a33aadf25b5.png)

```
<Canvas dpr={[1, 2]}>
      <ambientLight intensity={0.4}/>
      {markerList.map((lump, index) => (
          <>
              <Marker
                  key={index}
                  name='Marker'
                  position={[lump.posX, lump.posY, lump.posZ]}
              />
          </>
      ))}
      <DogModel/>
      <mesh
          onDoubleClick={e => {
          }}
          scale={[2, 200, 2.8]}
          position={[-0.9, 4, 0]}
          visible={true}
      >
          <boxGeometry/>
          <meshPhongMaterial color='royalblue' transparent opacity={0}/>
      </mesh>
      <PerspectiveCamera
          makeDefault
          ref={camera}
          position={[200, 0, 0]}
          zoom={window.innerWidth > 640 ? 15 : 10}
          fov={40}
      />
      <OrbitControls
          ref={orbit}
          target={[0, 1, 0]}
          enableRotate={true}
          minDistance={7}
          maxDistance={100}
          minPolarAngle={0}
          maxPolarAngle={Math.PI / 1.75}
          touches={{
              ONE: THREE.TOUCH.ROTATE,
              TWO: THREE.TOUCH.DOLLY_PAN
          }}
      />
    <Environment files="/old_depot_2k.hdr" ground={{height: '100%', radius: 130}}/>
</Canvas>
```

marker code is below

```
<mesh name={name}>
    <Html sprite position={[position[0], position[1], position[2]]}>
        <div
            id={'marker'}
            style={{display: status ? 'block' : 'none'}}
        >
            <Button.Ripple
                onPointerUp={setButtonAction}
                color='primary'
                className='btn-icon rounded-circle'>
                    <span className='TextWhite'>
                        <Plus size={16}/>
                    </span>
            </Button.Ripple>
        </div>
    </Html>
</mesh>
```

Thanks in advance

**Comment by isaac-mason** [comment, marked as answer]
You can use the `occlude` prop on `<Html />` for simple raycast occlusion.

See this drei storybook: https://drei.pmnd.rs/?path=/story/misc-html--html-raycast-occluder-st

**Reply by ekaterina4952** [reply]
Thanks for your reply. But I still have same problem.
If `occlude = true`, I don't see any markers. If `false`, It is same before.

**Reply by isaac-mason** [reply]
@ekaterina4952 if you're able to share a codesandbox I'm happy to take a look.

**Reply by ekaterina4952** [reply]
The `occlude ` doesnt works anymore in sandbox.
https://codesandbox.io/s/blissful-solomon-q41y1h?file=/src/App.js

**Reply by isaac-mason** [reply]
Fixed your codesandbox - https://codesandbox.io/s/onocclude-kzy05t

There were 2 issues:

1) The marker was positioned inside the model, so it was always occluded. I moved the marker outside of the model.
2) When only an `occlude` prop is provided, the children passed to `<Html />` are hidden by changing the opacity. When an `onOcclude` prop is passed in, this default is overridden and it's up to you to animate/handle the transition. I removed the `onOcclude` handler that was just doing a console.log

**Reply by ekaterina4952** [reply]
Thanks for your help

**Comment by isaac-mason** [comment]
Better support for occlusion is coming soon too! See https://github.com/pmndrs/drei/pull/1168

**Reply by ekaterina4952** [reply]
It is same before even if I use `occlude = 'raycast'` and `occlude = 'blending'`
